ocs_ci/ocs/node.py

```diff
@@ -908,6 +914,93 @@ def delete_and_create_osd_node_vsphere_upi(osd_node_name, use_existing_node=Fals
     return new_node_name
 
 
+def delete_and_create_osd_node_vsphere_lso(osd_node_name, use_existing_node=False):
+    """
+    Unschedule, drain and delete osd node, and creating a new osd node.
+    At the end of the function there should be the same number of osd nodes as
+    it was in the beginning, and also ceph health should be OK.
+    This function is for vSphere UPI.
+
+    Args:
+        osd_node_name (str): the name of the osd node
+        use_existing_node (bool): If False, create a new node and label it.
+            If True, use an existing node to replace the deleted node
+            and label it.
+
+    Returns:
+        str: The new node name
+
+    """
+    from ocs_ci.ocs.platform_nodes import PlatformNodesFactory
+    from ocs_ci.ocs.resources.storage_cluster import get_osd_size
+
+    sc_name = constants.LOCAL_BLOCK_RESOURCE
+    old_pv_objs = get_pv_objs_in_sc(sc_name)
+
+    osd_node = get_node_objs(node_names=[osd_node_name])[0]
+    osd_pod = get_node_pods(osd_node_name, pods_to_search=pod.get_osd_pods())[0]
+    osd_id = pod.get_osd_pod_id(osd_pod)
+    log.info(f"osd id to remove = {osd_id}")
+    # Save the node hostname before deleting the node
+    osd_node_hostname_label = get_node_hostname_label(osd_node)
+
+    log.info("Scale down node deployments...")
+    scale_down_deployments(osd_node_name)
+    log.info("Scale down deployments finished successfully")
+
+    new_node_name = delete_and_create_osd_node_vsphere_upi(
+        osd_node_name, use_existing_node
+    )
+    assert new_node_name, "Failed to create a new node"
+    log.info(f"New node created successfully. Node name: {new_node_name}")
+
+    # If we use LSO, we need to create and attach a new disk manually
+    new_node = get_node_objs(node_names=[new_node_name])[0]
+    plt = PlatformNodesFactory()
+    node_util = plt.get_nodes_platform()
+    osd_size = get_osd_size()
+    log.info(
+        f"Create a new disk with size {osd_size}, and attach to node {new_node_name}"
+    )
+    node_util.create_and_attach_volume(node=new_node, size=osd_size)
+
+    new_node_hostname_label = get_node_hostname_label(new_node)
+    log.info(
+        "Replace the old node with the new worker node in localVolumeDiscovery and localVolumeSet"
+    )
+    res = add_new_node_to_lvd_and_lvs(
+        old_node_name=osd_node_hostname_label,
+        new_node_name=new_node_hostname_label,
+    )
+    assert res, "Failed to add the new node to LVD and LVS"
+
+    log.info("Verify new pv is available...")
+    is_new_pv_available = verify_new_pv_available_in_sc(old_pv_objs, sc_name)
+    assert is_new_pv_available, "New pv is not available"
+    log.info("Finished verifying that the new pv is available")
+
+    switch_to_project(defaults.ROOK_CLUSTER_NAMESPACE)
```

Comment:
why is this needed?
Generally, I think that switching to a project, rather than specifying the namespace in the `oc` commands, is quite risky, because the test can fail before the switch back to the previous project took place, which means that the rest of the execution will be while the project is wrong

Reply:
Yes, I get your point. This step is mentioned in the docs: `oc project openshift-storage`, 
so I am not sure if we can remove it. @PrasadDesala @prsurve, what do you think?

Reply:
this is required to make sure ocs-removal job run's on openshift-storage and not in other ns

Reply:
maybe in the teardown part of the test, we can switch back to the default project

Reply:
why can't we use `-n openshift-storage` in the oc commands instead of switching to a project?

Reply:
Do you mean to change it just here in the test or also in the docs? We don't have to use `switch_to_project` in the test. But then we will have a tiny diff from the steps in the doc.

Reply:
change just here. I don't think that this diff is important

Reply:
Sure, np

Reply:
change just here. I don't think that this diff is important (docs wise)